firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloader.java

```diff
@@ -121,34 +121,114 @@ public static FirebaseModelDownloader getInstance(@NonNull FirebaseApp app) {
   public Task<CustomModel> getModel(
       @NonNull String modelName,
       @NonNull DownloadType downloadType,
-      @Nullable CustomModelDownloadConditions conditions)
-      throws Exception {
-    CustomModel localModel = sharedPreferencesUtil.getCustomModelDetails(modelName);
-    if (localModel == null) {
+      @Nullable CustomModelDownloadConditions conditions) {
+    CustomModel localModelDetails = getLocalModelDetails(modelName);
+    if (localModelDetails == null) {
       // no local model - get latest.
       return getCustomModelTask(modelName, conditions);
     }
 
     switch (downloadType) {
       case LOCAL_MODEL:
-        return Tasks.forResult(localModel);
+        return getCompletedLocalCustomModelTask(localModelDetails);
       case LATEST_MODEL:
         // check for latest model, wait for download if newer model exists
-        return getCustomModelTask(modelName, conditions, localModel.getModelHash());
+        return getCustomModelTask(modelName, conditions, localModelDetails.getModelHash());
       case LOCAL_MODEL_UPDATE_IN_BACKGROUND:
-        // start download in back ground, return local model
-        getCustomModelTask(modelName, conditions, localModel.getModelHash());
-        return Tasks.forResult(localModel);
+        // start upload in background, if newer model exists
```

Comment:
is it upload?

Reply:
no download - I fixed this in a bunch of places, where I was using it incorrectly. Thanks for noticing.